be/src/exec/json_scanner.cpp

```diff
@@ -717,28 +700,84 @@ Status JsonReader::_read_and_parse_json() {
         }
     }
 
-    data = reinterpret_cast<uint8_t*>(_parser_buf->ptr);
-    length = _parser_buf->remaining();
+    _payload_buffer.reset(_parser_buf->ptr);
+    _parser_buf->flip();
+    _payload_buffer_size = _parser_buf->remaining();
+    _payload_buffer_capacity = _parser_buf->remaining() + simdjson::SIMDJSON_PADDING;
+    return Status::OK();
+}
+
+// read one json string from file read and parse it to json doc.
+Status JsonReader::_read_file_broker() {
+    // TODO: Remove the down_cast, should not rely on the specific implementation.
+    auto* stream = down_cast<io::SeekableInputStream*>(_file->stream().get());
+    auto res = stream->get_size();
+    if (!res.ok()) {
+        return res.status();
+    }
+    auto sz = res.value();
```

Comment:
Does the sz means the whole file size? If the file size is huge, will it consume much memory?

Reply:
Yes. The memory usage would be as large as the input size.  It's necessary since the json parser we used does not support parsing in streaming mode. To eliminate this effect, the buffer is reused between json files.